builtin/game/chatcommands.lua

```diff
@@ -367,35 +369,35 @@ core.register_chatcommand("teleport", {
 			return pos, false
 		end
 
-		local teleportee = nil
 		local p = {}
 		p.x, p.y, p.z = string.match(param, "^([%d.-]+)[, ] *([%d.-]+)[, ] *([%d.-]+)$")
-		p.x = tonumber(p.x)
-		p.y = tonumber(p.y)
-		p.z = tonumber(p.z)
 		if p.x and p.y and p.z then
+			p.x = tonumber(p.x)
+			p.y = tonumber(p.y)
+			p.z = tonumber(p.z)
```

Comment:
You need to adjust `string.match` to be more accurate to guarantee this function call does not return `nil`. Simplest solution would again be to move it above the condition.